react-native-game-engine.d.ts

```diff
@@ -38,20 +39,32 @@ declare module "react-native-game-engine" {
     }
   
     export type GameEngineSystem = (entities: any, update: GameEngineUpdateEventOptionType) => any;
+    
+    interface GameEngineEntity {
+      [key:string]: any;
+      renderer?: JSX.Element | React.ComponentClass<any. any>;
+    }
 
+    type GameEngineEntities = Record<string | number, GameEngineEntity>;
+    
     export interface GameEngineProperties {
       systems?: any[];
       entities?: {} | Promise<any>;
       renderer?: any;
```

Comment:
Hey @SimpleProgrammingAU - this all looks really good. Quick question, should this line be: `renderer?: DefaultRenderer | any;` ?

Reply:
It should probably be `renderer: Renderer` and change the type definition for `DefaultRenderer` to `Renderer` to make it a bit more inclusive.

Reply:
I see.. I ask because of this change that you made: `timer?: DefaultTimer | any;` - hence I was thinking we do the same for the renderer for the sake of consistency..

What exactly did you mean by:

> and change the type definition for DefaultRenderer to Renderer to make it a bit more inclusive.

Are you saying that the `Default` in `DefaultRenderer` is redundant from a naming convention point of view? As in, the `DefaultRenderer` function should simple adhere to a `Renderer` function signature?

Lastly, since I'm not 100% across TS, would changing the name break anything from a backwards compatibility, tooling and intellisense (VS Code?) perspective?

Reply:
Let me have a play around with somethings myself because I'm not 100%, either! I'll get back to you tomorrow (I'm UTC+11) with either a new commit or more commentary =)

Reply:
@SimpleProgrammingAU I'm curious where this ended up. Any updates on your findings?